### PR TITLE
fix(ad_service_state_monitor): modify build error in rolling

### PR DESCRIPTION
**Pull request by wep21**
## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: system/ad_service_state_monitor/CMakeLists.txt [hunk 1]
```diff
@@ -29,6 +29,11 @@ ament_auto_add_executable(ad_service_state_monitor
   ${AD_SERVICE_STATE_MONITOR_SRC}
 )
 
+# workaround to allow deprecated header to build on both galactic and rolling
+target_compile_options(ad_service_state_monitor PRIVATE
```
Comment (esteve) [inline system/ad_service_state_monitor/CMakeLists.txt:33]: Please apply the same solution as in https://github.com/autowarefoundation/autoware.universe/pull/720#discussion_r852040761

**Review by esteve — CHANGES_REQUESTED** [review]

File: system/ad_service_state_monitor/CMakeLists.txt (hunk 1 above)
Comment (wep21) [reply system/ad_service_state_monitor/CMakeLists.txt]: @esteve addressed at https://github.com/autowarefoundation/autoware.universe/pull/716/commits/0fa477b3189323fd19c2b2f08542ffb398acee68.

File: system/ad_service_state_monitor/src/ad_service_state_monitor_node/state_machine.cpp [hunk 2]
```diff
@@ -12,13 +12,20 @@
...
 
+#include "ad_service_state_monitor/state_machine.hpp"
+
 #include <deque>
 #include <vector>
 
-#define FMT_HEADER_ONLY
-#include "ad_service_state_monitor/state_machine.hpp"
-
+#define FMT_HEADER_ONLY  // NOLINT
```
Comment (kenji-miyake) [inline system/ad_service_state_monitor/src/ad_service_state_monitor_node/state_machine.cpp:20]: Just out of curiosity, what's this `NOLINT` for?

**Review by kenji-miyake — APPROVED** [review]

File: system/ad_service_state_monitor/src/ad_service_state_monitor_node/state_machine.cpp (hunk 2 above)
Comment (wep21) [reply system/ad_service_state_monitor/src/ad_service_state_monitor_node/state_machine.cpp:20]: @kenji-miyake pre-commit fixes include order as below.
```
#define FMT_HEADER_ONLY
#include "ad_service_state_monitor/state_machine.hpp"

#include <fmt/format.h>
```
I felt this order is a little bit weird, so I added `NOLINT`

Comment (kenji-miyake) [reply system/ad_service_state_monitor/src/ad_service_state_monitor_node/state_machine.cpp:20]: I see, but I think it's okay to define all macros at the top of headers.

**Pull request merged**